Mathlib/Topology/Algebra/ProperAction.lean

```diff
@@ -0,0 +1,309 @@
+  then the quotient space is Hausdorff (T2).
+* `t2Space_of_properSMul_of_t2Group`: If a T2 group acts properly on a topological space,
+  then this topological space is T2.
+* `ProperlyDiscontinuousSMul_iff_ProperSMul`: If a discrete group acts on a T2 space `X` such that
+  `X × X` is compactly generated, then the action is properly discontinuous if and only if it is
+  continuous in the second variable and proper. This in particular true if `X` is locally compact
+  or first-countable.
+
+## Implementation notes
+
+Concerning `ProperlyDiscontinuousSMul_iff_ProperSMul`, this result should be the only one needed
+to link properly discontinuous and proper actions, however the notion of compactly generated space
+is not yet in Mathlib so it is used as an intermediate result to prove
+`WeaklyLocallyCompactSpace.isProperMap_iff_isCompact_preimage` and
+`SequentialSpace.isProperMap_iff_isCompact_preimage`.
+
+## References
+
+* [N. Bourbaki, *General Topology*][bourbaki1966]
+
+## Tags
+
+Hausdorff, group action, proper action
+-/
+
+open Filter Topology Set Prod
+
+/-- Proper group action in the sense of Bourbaki:
+the map `G × X → X × X` is a proper map (see `isProperMap`). -/
+@[mk_iff]
+class ProperVAdd (G X : Type*) [TopologicalSpace G] [TopologicalSpace X] [AddGroup G]
+    [AddAction G X] : Prop where
+  /-- Proper group action in the sense of Bourbaki:
+  the map `G × X → X × X` is a proper map (see `isProperMap`). -/
+  isProperMap_vadd_pair' : IsProperMap (fun gx ↦ ⟨gx.1 +ᵥ gx.2, gx.2⟩ : G × X → X × X)
+
+/-- Proper group action in the sense of Bourbaki:
+the map `G × X → X × X` is a proper map (see `isProperMap`). -/
+@[to_additive existing, mk_iff]
+class ProperSMul (G X : Type*) [TopologicalSpace G] [TopologicalSpace X] [Group G]
+    [MulAction G X] : Prop where
+  /-- Proper group action in the sense of Bourbaki:
+  the map `G × X → X × X` is a proper map (see `isProperMap`). -/
+  isProperMap_smul_pair' : IsProperMap (fun gx ↦ ⟨gx.1 • gx.2, gx.2⟩ : G × X → X × X)
+
+attribute [to_additive existing] properSMul_iff
+
+/-- By definition, if G acts properly on X then
+the map `G × X → X × X` is a proper map. -/
+@[to_additive "By definition, if G acts properly on X then
+the map `G × X → X × X` is a proper map."]
+lemma isProperMap_smul_pair (G X : Type*) [Group G] [MulAction G X]
+    [TopologicalSpace G] [TopologicalSpace X] [ProperSMul G X] :
+    IsProperMap (fun gx ↦ ⟨gx.1 • gx.2, gx.2⟩ : G × X → X × X) :=
+  ProperSMul.isProperMap_smul_pair'
+
+variable {G X Y Z : Type*} [Group G] [MulAction G X] [MulAction G Y]
+variable [TopologicalSpace G] [TopologicalSpace X] [TopologicalSpace Y]
+variable [TopologicalSpace Z]
+
+/-- If a group acts properly then in particularl it acts continuously. -/
+@[to_additive "If a group acts properly then in particularl it acts continuously."]
+instance continuousSmul_of_properSMul [ProperSMul G X] : ContinuousSMul G X where
```

Comment:
```suggestion
-- See note [lower instance priority]
instance (priority := 100) continuousSmul_of_properSMul [ProperSMul G X] : ContinuousSMul G X where
```
since it always applies (grep to find the note and read it)